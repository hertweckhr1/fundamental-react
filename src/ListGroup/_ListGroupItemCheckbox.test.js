/* eslint-disable jsx-a11y/anchor-is-valid */
import ListGroup from './ListGroup';
import ListGroupItemCheckbox from './_ListGroupItemCheckbox';
import { mount } from 'enzyme';
import React from 'react';
import renderer from 'react-test-renderer';

describe('<ListGroupItemCheckbox />', () => {
    const listGroupCheckbox = (
        <ListGroup>
            <ListGroup.Item>
                <ListGroupItemCheckbox>List item 1</ListGroupItemCheckbox>
            </ListGroup.Item>
        </ListGroup>
    );

    test('create list group with checkbox', () => {
        // list group with checkbox
        let component = renderer.create(listGroupCheckbox);
        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });

    describe('Prop spreading', () => {
        test('should allow props to be spread to the ListGroupItemCheckbox component', () => {
            // TODO: placeholder for this test description once that functionality is built
            const element = mount(<ListGroupItemCheckbox data-sample='Sample' />);

            expect(
                element.getDOMNode().attributes['data-sample'].value
            ).toBe('Sample');
        });

        test('should allow props to be spread to the ListGroupItemCheckbox component\'s input element', () => {
            const element = mount(<ListGroupItemCheckbox inputProps={{ 'data-sample': 'Sample' }} />);

            expect(
                element.find('input').getDOMNode().attributes['data-sample'].value
            ).toBe('Sample');
        });

        test('should allow props to be spread to the ListGroupItemCheckbox component\'s label element', () => {
            const element = mount(<ListGroupItemCheckbox labelProps={{ 'data-sample': 'Sample' }} />);

            expect(
                element.find('label').getDOMNode().attributes['data-sample'].value
            ).toBe('Sample');
        });
    });
});
