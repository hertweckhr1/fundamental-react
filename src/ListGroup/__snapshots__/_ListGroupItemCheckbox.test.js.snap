// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ListGroupItemCheckbox /> create list group with checkbox 1`] = `
<ul
  className="fd-list-group"
>
  <li
    className="fd-list-group__item"
  >
    <div
      className="fd-form__item fd-form__item--check"
    >
      <label
        className="fd-form__label"
        htmlFor="CndSd399"
      >
        <input
          className="fd-form__control"
          id="CndSd399"
          type="checkbox"
        />
        List item 1
      </label>
    </div>
  </li>
</ul>
`;
